.github/CONTRIBUTING.md

```diff
@@ -1,38 +1,148 @@
-require no new tests. If you are adding functionality or fixing a bug, we need
-a test!
+3. Run the tests. We only take pull requests with passing tests, and it's great to know that you have a clean slate: `ember exam`. More information on testing commands can be found [here](../docs/USAGE.md#running-tests)
 
-4. Make the test pass.
+Okay, you're ready to go!
 
-5. Push to your fork and submit a pull request.
+## How to tackle a new feature
+
+If there's already an issue for the feature you want to tackle, how complete is the description? Do you know what to do next? Are you confident you know what "done" looks like?
+
+If there's not an issue yet, write one.
+
+Whether you're writing a new issue or improving on an existing issue, be sure to clarify exactly how you expect the finished change to look and work.
+
+One of the best ways to write a feature is in user story format:
+
+> As an [actor], I want to do [action], so that [benefit].
+
+For example, let's say we wanted to write some Slack integration for new comments posted to a Code Corps project. That user story might look something like:
+
+> As a project maintainer, I want to see new comments on my project task in my Slack channel so that everyone can see and react to some of the latest changes in the project.
+
+You may want to go deeper into detail. Posting screenshots of designs or expected test cases and scenarios are even more helpful. Place yourself in the shoes of the person who's going to accomplish the task – even if that person is you. What steps should I be taking next to finish this task?
+
+Once you've created the issue, you can [make your changes and push them up](#how-to-write-new-code).
+
+## What kind of documentation are you writing?
+
+- [I want to document the codebase](#improving-ember-docs).
+- [I want to improve the docs on GitHub.](#improving-the-readme)
+
+### Improving Ember docs
+
+You can see how much code is documented on [Inch CI](http://inch-ci.org/github/code-corps/code-corps-ember).
+
+We use YUIDoc to document our code.
+
+You can learn about how to write documentation with YUIDoc from these places:
+
+- Our [Styleguide](../docs/STYLEGUIDE.md#Documentation)
+- YUIDoc's [syntax guide](http://yui.github.io/yuidoc/syntax/index.html)
+
+[Done with your changes?](#i-finished-my-changes)
+
+### Improving the README
+
+If you're just looking to improve the README, there's a couple things you should know:
+
+- Open an issue first. It's better if we discuss your proposed changes.
+- We try to keep the main `README.md` lightweight and use it as a jumping point to other docs.
+- Most other docs can be placed in `/docs`.
+- Try to make it easy for people to jump around in your doc.
+
+[Done with your changes?](#i-finished-my-changes)
+
+## How to fix a bug
+
+If you're fixing a bug that's already been added to the issues, ask yourself whether the bug description is clear? Do you know what circumstances led to the bug? Does it seem easy to reproduce?
```

Comment:
Should? or shouldn't? I think it should be:
>...ask yourself whether the bug description is clear. Do you know what circumstances led to the bug?...

If we want it to be a question then it'll need a rewrite.